sample/src/main/kotlin/com/adevinta/spark/sample/BrowserApp.kt

```diff
@@ -0,0 +1,417 @@
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.navigation.NavGraphBuilder
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.currentBackStackEntryAsState
+import androidx.navigation.compose.rememberNavController
+import com.adevinta.spark.ExperimentalSparkApi
+import com.adevinta.spark.SparkTheme
+import com.adevinta.spark.components.appbar.TopAppBar
+import com.adevinta.spark.components.icons.Icon
+import com.adevinta.spark.components.icons.IconButton
+import com.adevinta.spark.components.scaffold.Scaffold
+import com.adevinta.spark.components.text.Text
+import com.adevinta.spark.components.textfields.TextField
+import com.adevinta.spark.icons.SparkIcon
+import com.adevinta.spark.tokens.darkSparkColors
+import com.adevinta.spark.tokens.lightSparkColors
+import com.airbnb.android.showkase.models.ShowkaseBrowserComponent
+import com.airbnb.android.showkase.models.ShowkaseCategory
+import com.airbnb.android.showkase.ui.SemanticsUtils.lineCountVal
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+internal fun ShowkaseBrowserApp(
+    groupedComponentMap: Map<String, List<ShowkaseBrowserComponent>>,
+    showkaseBrowserScreenMetadata: MutableState<ShowkaseBrowserScreenMetadata>,
+) {
+    val colors = if (isSystemInDarkTheme()) darkSparkColors() else lightSparkColors()
+    SparkTheme(colors = colors) {
+        CompositionLocalProvider(
+            LocalInspectionMode provides true,
+        ) {
+            val navController = rememberNavController()
+            val scrollBehavior = TopAppBarDefaults.pinnedScrollBehavior()
+            Scaffold(
+                modifier = Modifier.nestedScroll(scrollBehavior.nestedScrollConnection),
+                topBar = {
+                    AppBar(
+                        navController = navController,
+                        scrollBehavior = scrollBehavior,
+                        showkaseBrowserScreenMetadata = showkaseBrowserScreenMetadata,
+                    )
+                },
+                content = {
+                    BodyContent(
+//                        modifier = Modifier.fillMaxSize(),
+                        contentPadding = it,
+                        navController = navController,
+                        groupedComponentMap = groupedComponentMap,
+                        showkaseBrowserScreenMetadata = showkaseBrowserScreenMetadata,
+                    )
+                },
+            )
+        }
+    }
+}
+
+@OptIn(ExperimentalMaterial3Api::class, ExperimentalSparkApi::class)
+@Composable
+internal fun AppBar(
+    navController: NavHostController,
+    scrollBehavior: TopAppBarScrollBehavior,
+    showkaseBrowserScreenMetadata: MutableState<ShowkaseBrowserScreenMetadata>,
```

Comment:
nit: let's use simpler variable name, like `metadata`. It makes the code more readable.